### ISSUE TITLE
Wrong redirect when setting location.href=name from application code

### ISSUE THREAD

**Issue by bahmutov** [cypress-io/cypress#5099]
Given `cypress.json` with the following

```json
{
  "baseUrl": "https://www.cypress.io/features"
}
```

I am loading the page, and then the application redirects to new _relative_ page using `location.href = 'bar'` syntax.

```js
describe('page', () => {
  it('works', () => {
    cy.visit('/')
    cy.log('setting href to bar')
    cy.window().then(w => {
      w.location.href = 'bar'
    })
  })
})
```

## Expected behavior

It should go to 404 page with url `https://www.cypress.io/bar` just like the screenshot shows 

<img width="1399" alt="Screen Shot 2019-09-07 at 12 40 18 AM" src="https://user-images.githubusercontent.com/2212006/64469891-290a5f80-d108-11e9-9be7-376b51bfa222.png">

## Behavior in Chrome

Chrome 76, 78 shows the following with the top url WRONG ``https://www.cypress.io/__/bar`

<img width="1218" alt="Screen Shot 2019-09-07 at 12 41 44 AM" src="https://user-images.githubusercontent.com/2212006/64469904-5951fe00-d108-11e9-8b51-01244a4c77c8.png">

## Reproduction

Demo: branch `test-page` in https://github.com/cypress-io/cypress-test-tiny/tree/test-page

Weird behavior for another day: Electron does not load the base url at all.

<img width="1282" alt="Screen Shot 2019-09-07 at 12 38 45 AM" src="https://user-images.githubusercontent.com/2212006/64469879-e21c6a00-d107-11e9-9bfb-7774d7c1afcd.png">

## Really wrong behavior

Especially noticeable when the app goes from one page to another. For example this test

```js
describe('page', () => {
  it('works', () => {
    cy.visit('https://docs.cypress.io/guides/overview/why-cypress.html')
    cy.log('setting href to bar')
    cy.window().then(w => {
      w.location.href = 'writing-and-organizing-tests.html'
    })
  })
})
```

In regular browser it just goes to the new page

<img width="811" alt="Screen Shot 2019-09-07 at 12 46 25 AM" src="https://user-images.githubusercontent.com/2212006/64469947-1b090e80-d109-11e9-8a3d-a8eb2f3e343e.png">

In Chrome under Cypress it inserts `/__/` in place of `/guides/core-concepts/` which breaks the navigation

<img width="1219" alt="Screen Shot 2019-09-07 at 12 48 12 AM" src="https://user-images.githubusercontent.com/2212006/64469962-455acc00-d109-11e9-96a7-35df1adc9f3e.png">

**Comment by jennifer-shehane** [comment]
### BROKEN IN: 
- Cypress v2.1.0 ~ v3.4.1
- Chrome Canary 79.0.3906.0 
- Chrome 76.0.3809.132
- Chrome 74.0.3729.0
- Chrome 71.0.3572.0
- Electron 61

I'm not sure why we wanted to check various Chrome versions - as if this ever worked 🤔

**Comment by jennifer-shehane** [comment]
Duplicate of #3975  which is currently being fixed in https://github.com/cypress-io/cypress/pull/5273

Please see #3975 for any followup.

**Comment by jennifer-shehane** [comment]
This fix is available starting in [`4.6.0`](https://docs-staging.cypress.io/guides/references/changelog.html#4-6-0) as an [experiment](https://on.cypress.io/experiments) which you can access by setting this config option in your `cypress.json` or elsewhere:

```json
{
	"experimentalSourceRewriting": true
}
```

The fix is experimental, so there may be some situations where the this is not fixed. 

If you're still this issue while setting the `experimentalSourceRewriting` to true in 4.6.0 - [open a new issue](https://github.com/cypress-io/cypress/issues/new) with a reproducible example + screenshots, etc - filling out our issue template.
